test/parallel/test-util-callbackify.js

```diff
@@ -225,3 +225,37 @@ const values = [
     })
   );
 }
+
+{
+  // Verify that non-function inputs throw.
+  ['foo', null, undefined, false, 0, {}, Symbol(), []].forEach((value) => {
+    assert.throws(() => {
+      callbackify(value);
+    }, common.expectsError({
+      code: 'ERR_INVALID_ARG_TYPE',
+      type: TypeError,
+      message: 'The "original" argument must be of type function'
+    }));
+  });
+}
+
+{
+  async function asyncFn() {
+    return await Promise.resolve(42);
```

Comment:
Indeed, but that's the way to other tests are written (e.g. https://github.com/nodejs/node/blob/master/test/parallel/test-util-callbackify.js#L31)
The rationale was to make them actually `async` in timing, not just in signature.